src/helpers/verifyToken.ts

```diff
@@ -0,0 +1,15 @@
+import { verify } from 'jsonwebtoken'
+import config from '../config/environment'
+
+const verifyToken = (token:string):Promise<any> =>
+  new Promise((resolve, reject) => {
+    verify(token, config.secretKey || 'secret', (err, decoded) => {
```

Comment:
secretKey should be always required and shouldn't be `string | undefined`.
A way to ensure that in `config/environment.ts` you can add for example
`if (!SECRET_KEY) {
 throw new Error('SECRET_KEY is not provided in env vars')
}
`
to be something like this
 ```
if (!connectionString) {
  throw new Error('Database url is not valid')
}

if (!SECRET_KEY) {
  throw new Error('SECRET_KEY is not provided in env vars')
}

const config = {
  nodeEnv: NODE_ENV,
  port: PORT || 8080,
  connectionString,
  ssl,
  secretKey: SECRET_KEY
}

export default config
```
```suggestion
    verify(token, config.secretKey, (err, decoded) => {
```